### PR TITLE
SDIT-1410 UserResource

### PR DESCRIPTION
**Pull request by steverendell**
switch some /users endpoints to client token

### CODE REVIEW

File: backend/middleware/currentUser.test.ts [hunk 1]
```diff
@@ -2,39 +2,37 @@ import * as jwt from 'jsonwebtoken'
...
-    prisonApi.userCaseLoads = jest.fn()
-    // @ts-expect-error ts-migrate(2339) FIXME: Property 'setActiveCaseload' does not exist on typ... Remove this comment to see the full error message
-    prisonApi.setActiveCaseload = jest.fn()
-    // @ts-expect-error ts-migrate(2339) FIXME: Property 'currentUser' does not exist on type '{}'... Remove this comment to see the full error message
-    hmppsManageUsersApi.currentUser = jest.fn()
-
-    // @ts-expect-error ts-migrate(2339) FIXME: Property 'currentUser' does not exist on type '{}'... Remove this comment to see the full error message
-    hmppsManageUsersApi.currentUser.mockReturnValue({
+    jest.resetAllMocks()
+    hmppsManageUsersApi.currentUser.mockResolvedValue({
```
Comment (steverendell) [inline backend/middleware/currentUser.test.ts:19]: Tidy up FIXMEs etc

File: backend/controllers/changeCaseload.ts [hunk 2]
```diff
@@ -47,7 +50,3 @@ export const changeCaseloadFactory = (prisonApi, logError) => {
 
   return { index, post }
 }
-
-export default {
-  changeCaseloadFactory,
-}
```
Comment (steverendell) [inline backend/controllers/changeCaseload.ts:53]: simplify export

File: backend/controllers/establishmentRoll/noCellAllocated.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { Request, Response } from 'express'
 import { formatName, putLastNameFirst, getTime, stripAgencyPrefix } from '../../utils'
 
 export default ({ oauthApi, systemOauthClient, prisonApi }) =>
-  async (req: Request, res: Response) => {
+  async (req: Partial<Request>, res: Partial<Response>) => {
```
Comment (steverendell) [inline backend/controllers/establishmentRoll/noCellAllocated.ts:5]: Add types where possible

File: backend/tests/attendanceChanges.test.ts [hunk 4]
```diff
@@ -15,30 +15,35 @@ describe('Attendance change router', () => {
...
+  const systemOauthClient = {
+    getClientCredentialsTokens: jest.fn(),
+  }
+  const systemToken = { token: 'some-token' }
+
+  const req = {
+    originalUrl: 'http://localhost',
+    query: {},
+    session: { userDetails: { username: 'ME' } },
+  }
```
Comment (steverendell) [inline backend/tests/attendanceChanges.test.ts:27]: Move to consts where possible

File: backend/tests/establishmentRoll/noCellAllocated.test.ts [hunk 5]
```diff
@@ -189,7 +185,7 @@ describe('No cell allocated', () => {
       const error = new Error('Network error')
       prisonApi.getInmatesAtLocationPrefix.mockRejectedValue(error)
 
-      await expect(controller(req, res)).rejects.toThrowError(error)
+      await expect(controller(req, res)).rejects.toThrow(error)
```
Comment (steverendell) [inline backend/tests/establishmentRoll/noCellAllocated.test.ts:188]: switch deprecated method

File: src/Attendance/AttendanceOptions/AttendanceOptions.test.js [hunk 6]
```diff
@@ -65,7 +65,9 @@ describe('<AttendanceOptions />', () => {
   })
 
   it('should successfully call updateOffenderAttendance', () => {
-    act(() => getPayRadio().props.onChange())
+    act(() => {
+      getPayRadio().props.onChange()
+    })
```
Comment (steverendell) [inline src/Attendance/AttendanceOptions/AttendanceOptions.test.js:70]: Fix a warning regarding void callback

**Review by mikehalmamoj — DISMISSED** [review]

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**